### PR TITLE
Updated docker documentation, added .dockerignore

### PR DESCRIPTION
**Pull request by DreamingRaven**
## Description

TLDR: Did not change much, just changed documentation to run docker properly and added .dockerignore for good practice.

.dockerignore is always important to have similarly to .gitignore to
keep in this case the docker context free from unecessary bloat.

Updated the documentation to correct the docker build and run invocation
so that it keep the existing style of using a subdirectory to store the
dockerfiles but puts the context in the current parent directory so the
files can be copied in properly.

## Affected Dependencies
None

## How has this been tested?
- Describe the tests that you ran to verify your changes.
ran the new commands
```
sudo docker build -t tenseal . -f docker-images/Dockerfile-py38
```
and 
```
sudo docker run -it tenseal
```
- Provide instructions so we can reproduce.
please run the above two commands or:
```
sudo docker build -t tenseal . -f docker-images/Dockerfile-py38 && sudo docker run -it tenseal
```
- List any relevant details for your test configuration.

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

### CODE REVIEW

**Comment by DreamingRaven** [comment]
related to: https://github.com/OpenMined/TenSEAL/issues/72

File: .dockerignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+tutorials
+third_party
```
Comment (youben11) [inline .dockerignore:2]: I don't think those should be ignored, someone might want to run the docker image to test the tutorials which he expect to be there, third_party libraries also doesn't need to be downloaded if they are already on the host machine. Some example directories that might need to be ignored are build, bin, dist, lib, and wheels which might be generated in the working directory when installing and building the library.

Comment (DreamingRaven) [reply .dockerignore:2]: Ah ok yeah, I misunderstood what third party was, I will remove that line.

As for tutorials, they are data, image, and jupyter notebook files from the looks of it. However they have no way to run jupyter  files in your docker containers, nor does the container expose a port for users to interact with the jupyter server if it was running, so they could play around with these tutorials even if the files were included. 
I will remove that line if you still want the file in, but I don't think its reasonable to expect an average user to be able to modify the docker container to be able to run jupyter notebooks since the container does not support it, and they are unnecessary for everyone else to my understanding.

Comment (youben11) [reply .dockerignore]: True that we doesn't install juputer-notebook in the current images, maybe we should ignore them and build a specific image for that. Then we can leave it as is, but can you add [build, bin, dist, lib, wheels] also, those are the one I know about which are generated when I'm building and playing with the package, I don't want them to be included in the docker image.

Comment (DreamingRaven) [reply .dockerignore]: Yeah no problem will add them now.

Comment (youben11) [reply .dockerignore]: also, please merge master into your branch so that I can merge it back to master

Comment (DreamingRaven) [reply .dockerignore]: done

Comment (youben11) [reply .dockerignore]: Thank you!
